src/iDynFor/iDynTreePinocchioConversions.h

```diff
@@ -4,14 +4,143 @@
  */
 
 // pinocchio
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/algorithm/frames.hpp>
+#include <pinocchio/multibody/fwd.hpp>
 #include <pinocchio/spatial/inertia.hpp>
 
 // iDynTree 
 #include <iDynTree/Core/SpatialInertia.h>
+#include <iDynTree/Model/Model.h>
 
 namespace iDynFor
 {
+namespace details
+{
+
+// All this part is taken from https://github.com/stack-of-tasks/pinocchio/blob/v2.6.8/src/parsers/urdf/model.hxx
+// Unfortunatly, it is not public, so it can't be reused
+template<typename _Scalar, int Options>
+class iDynTreeVisitorBaseTpl {
```

Comment:
Nitpick, should it be ``iDynTree*`` or ``iDynFor*`` or even simply ``VisitorBaseTpl`` since it is already in the ``iDynFor`` namespace?

Reply:
Good point, this is named after [`UrdfVisitorBaseTpl`](https://github.com/stack-of-tasks/pinocchio/blob/280005c8d99c2485ee942b6a38c4dc0edf75c706/src/parsers/urdf/model.hxx#L26). It is a visitor for `ìDynTree::Model`, so probably a more descriptive name is `iDynTreeModelVisitorBaseTpl`, what do you think?

Reply:
Fixed in https://github.com/ami-iit/idynfor/pull/11/commits/d637ee459652386c8affa5d3a5207efcae0a2154 .

Reply:
Ah ok I see, I was not sure what it was visiting indeed. ``iDynTreeModelVisitorBaseTpl`` does make sense then